README.md

```diff
@@ -136,6 +136,85 @@ Pathname | Result
 /example | Home / Custom Example
 /custom-props | Home / Hi
 
+### Advanced (Declarative Routes)
+Same as the example above using Declarative Routing.
+
+```js
+import useBreadcrumbs from 'use-react-router-breadcrumbs';
+
+const userNamesById = { '1': 'John' }
+
+const DynamicUserBreadcrumb = ({ match }) => (
+  <span>{userNamesById[match.params.userId]}</span>
+);
+
+const CustomPropsBreadcrumb = ({ someProp }) => (
+  <span>{someProp}</span>
+);
+
+// define custom breadcrumbs for certain routes.
+// breadcumbs can be components or strings.
+
+// map & render your breadcrumb components however you want.
+const BreadcrumbTrail = (routes) => {
+  const breadcrumbs = useBreadcrumbs(routes);
+
+  return (
+    <>
+      {breadcrumbs.map(({
+        match,
+        breadcrumb
+      }) => (
+        <span key={match.pathname}>
+          <NavLink to={match.pathname}>{breadcrumb}</NavLink>
+        </span>
+      ))}
+    </>
+  );
+};
+
+const BuildAppRoutes = () => {
+  // Full declarative react router support. example: Element, children, Nested Routes
+  return (
+    <Route path='/users/:userId' breadcrumb={DynamicUserBreadcrumb} element={<ProfilePage/>} />
+    <Route path='/example' breadcrumb='Custom Example' >
+      <Route path='/' breadcrumb='Custom Example' >
+        <ExamplePage/>
+      </Route>
+    </Route>
+    <Route path='/custom-props' breadcrumb={CustomPropsBreadcrumb} props={ someProp: 'Hi' } >
+      <CustomPage/>
+    </Route>
+  )
+};
+
+const AppRouter = () => {
+  // You could use context to set app Routes and add the breadcrumbs somewhere deeper in the application layout.
+  const BreadCrumbs = buildBreadcrumbs(routes)
```

Comment:
`routes` seems to be missing a definition here? At this point in the example I believe it would be `undefined`.